rust/lance/src/index/scalar.rs

```diff
@@ -465,6 +465,73 @@ pub fn index_matches_criteria(
     Ok(true)
 }
 
+/// Initialize a scalar index from a source dataset
+pub async fn initialize_scalar_index(
+    target_dataset: &mut Dataset,
+    source_dataset: &Dataset,
+    source_index: &lance_table::format::Index,
+    column_name: &str,
+) -> Result<()> {
+    use lance_index::metrics::NoOpMetricsCollector;
+    use lance_index::scalar::inverted::tokenizer::InvertedIndexParams;
```

Comment:
I find it kind of weird that we classify inverted indexes as "scalar". I'm pretty sure this code is just mirroring existing stuff but I think from an inverted index POV the values are collections instead of scalars and it'll be more of a stretch than it is today (with strings) if we ever extend our inverted indexes to cover JSON docs or arrays.

Maybe something we can clean up later if it bugs others.

Reply:
I think so far the definition of "scalar" is that it is working against column(s) that is not a vector column, so inverted index also fits in the definition. Agree it feels like one giant index category compared to other scalar index types, and the implementation is much more complicated than all the other ones. But so far the interfaces used seem pretty consistent.